packages/api/package.json

```diff
@@ -81,7 +81,7 @@
     "@web5/common": "1.0.0",
     "@web5/crypto": "1.0.0",
     "@web5/dids": "1.1.0",
-    "@web5/user-agent": "0.3.8"
+    "@web5/user-agent": "workspace:*"
```

Comment:
maybe it's just me but i find this impossible to reason about or manage. are we going to rely on lockfiles or hashes to know what versions went together? 

even when i look at the lockfile, the changed version isn't even tracked and there isn't even a hash.

Reply:
For local development: it will always point to the code on the local machine. The intent is that if a dependent package has a breaking change to downstream mono repo, CICD will not let the PR through.

For released package, you can checkout the commit of  `Version Packages` (either via command or GitHub UI), and see the version.

Reply:
For end-user importing a package or release, they'd know the dependency version the same way they alway would, if they care to know.

Reply:
> For local development: it will always point to the code on the local machine

right, that's not the problem with this though.

> For released package, you can checkout the commit of Version Packages (either via command or GitHub UI), and see the dependency versions.

Right, that's the problem getting at is that there will be no versioning of packages under this paradigm. You are sidestepping that flow here. The versions are not being checked into version control. You are opaquely publishing them to npm without accounting for them in version control and planning on pnpm to "take care of translating the workspace:* into the right version." Which does not get checked into version control, as evidenced by this PR. 

Maybe to clarify: is the intention for this to be a permanent change?

Reply:
> is the intention for this to be a permanent change?

Yes, I intended for this to be a permanent change. To be fair, I am a complete noob to `pnpm` that was introduced to `web5-js` repo, but the desire for wanting a mono repro is clear to me because the exact same problems existed in previous projects I worked on.

When `npm` publishes package A the depends B, it looks into A's `pacakge.json` for version of B to use.
When `pnpm` publishes package A the depends B, it looks into A's `package.json` for version of B to use, which points to the `package.json` of B that contains the version number.

It's merely difference between how two different tools publishes a package, though sure, `npm` is more "first class", but here we are choosing `pnpm` to augment the limitations of `npm`.

Reply:
I get all of that already and it's not the root of the problem I'm raising. Let me explain more clearly maybe. We have a CI flow where we release off of `main` using changesets. Each PR increments the appropriate version as long as you create a changeset. This is very predictable. There's no way to screw it up, and everyone is required to make a changeset as long as they're touching production code. Each version gets checked in. The change that was made cannot live alongside our current release process as it exists first because it will simply get replaced by the automation with real version numbers (and thus be pointless), second because we release off of main and don't have a bunch of branches that are storing the intermediate state of the repo at each release which means you can't just not track versions on main, and third because the consequences of not tracking versions would be that no one knows what's going on with the code that's being published on npm because it doesn't match the state of version control. So I could approve this but keep in mind it's just going to get replaced by our automation anyways, and doesn't work with our current release workflows.

Reply:
@shamilovtim, I apologize for my stupidity, I read your response above multiple times and still don't get the full thing, I get the main point being: `workspace:*` does not work with our current release workflows with 3 reasons. Assuming we just need one of them to be true to prove `workspace:*` is a bad idea, let's just look at the first one:

> it will simply get replaced by the automation with real version numbers

I don't think this is true (but can be proven wrong easily of course by just getting this merged and see what happens), it'd be odd to me for the changeset-bot create a PR that replaces `workspace:*` to specific versions, because that kinda defeats the point this feature... but of course would validate your point.

I don't fully understand your point 2 and 3, but can't say I agree based on my interpretation.

Reply:
No additional comments, so will merge and see what changeset-bot does.